src/main/java/com/nutomic/syncthingandroid/activities/ShareActivity.java

```diff
@@ -272,4 +341,40 @@ protected void onPostExecute(Boolean isError) {
             finish();
         }
     }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        if (mFoldersSpinner.getSelectedItem() != null) {
+            Folder selectedFolder = (Folder) mFoldersSpinner.getSelectedItem();
+            PreferenceManager.getDefaultSharedPreferences(this).edit()
+                    .putString(PREF_PREVIOUSLY_SELECTED_SYNCTHING_FOLDER, selectedFolder.id)
+                    .apply();
+        }
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == FolderPickerActivity.DIRECTORY_REQUEST_CODE && resultCode == RESULT_OK) {
+            Folder selectedFolder = (Folder) mFoldersSpinner.getSelectedItem();
+            String folderDirectory = selectedFolder.path;
+            //The folder paths are not entirely consistent in terms of what character they end with,
+            //that is why a / is added if it is missing so it can be used to cut the corresponding part
+            //from the subDirectory string, so only the sub directory is displayed.
+            if (!folderDirectory.endsWith("/")) {
+                folderDirectory += "/";
+            }
+
+            String subDirectory = data.getStringExtra(FolderPickerActivity.EXTRA_RESULT_DIRECTORY) + "/";
```

Comment:
Instead of messing around with path formatting manually, you can just use [`file.toURI().normalize().getPath()`](https://stackoverflow.com/a/21849556/1837158) to get the properly formatted path. You can put that in a method in `Util` and call it on FolderPickerActivity:200 and ShareActivity:361.